parts/k8s/windowsazurecnifunc.ps1

```diff
@@ -76,10 +76,25 @@ Set-AzureCNIConfig
     $configJson = Get-Content $fileName | ConvertFrom-Json
     $configJson.plugins.dns.Nameservers[0] = $KubeDnsServiceIp
     $configJson.plugins.dns.Search[0] = $KubeDnsSearchPath
-    $configJson.plugins.AdditionalArgs[0].Value.ExceptionList[0] = $KubeClusterCIDR
-    $configJson.plugins.AdditionalArgs[0].Value.ExceptionList[1] = $MasterSubnet
+
+    $osBuildNumber = (get-wmiobject win32_operatingsystem).BuildNumber
+    if ($osBuildNumber -le 17763){
+        # In WS2019 and below rules in the exception list are generated by dropping the prefix lenght and removing duplicate rules.
+        # If multiple execptions are specified with different ranges we should only include the broadest range for each address.
+        # This issue has been addressed in 19h1+ builds
+
+        $originalExceptions = @($KubeClusterCIDR, $MasterSubnet, $VNetCIDR)
+        $processedExpceptions = GetBroadestRangesForEachAddress $originalExceptions
+        Write-Host "Filtering CNI config exception list values to work aroudn WS2019 issue processing rules. Original exception list: $originalExceptions, processed exception list: $processedExpceptions"
+        $configJson.plugins.AdditionalArgs[0].Value.ExceptionList = $processedExpceptions
+    }
+    else {
+        $configJson.plugins.AdditionalArgs[0].Value.ExceptionList[0] = $KubeClusterCIDR

```

Comment:
I actually like putting all the exceptions in this array. I think it would clean this up a bit by setting
`$exceptions = @($KubeClusterCIDR, $MasterSubnet, $VNetCIDR)` once at the top, filtering it needed, then setting `$configJson.plugins.AdditionalArgs[0].Value.ExceptionList = $exceptions` in just one place. That way if we need to add more exceptions later it doesn't need to be done on each side of the if/else. What do you think?

Reply:
Yea, good idea!